source/simulator/assembly.cc

```diff
@@ -301,12 +360,32 @@ namespace aspect
 
         template <int dim>
         AdvectionSystem<dim>::
-        AdvectionSystem (const FiniteElement<dim> &finite_element)
+        AdvectionSystem (const FiniteElement<dim> &finite_element,
+                         const bool                field_is_discontinuous)
           :
           local_matrix (finite_element.dofs_per_cell,
                         finite_element.dofs_per_cell),
+          local_matrices_int_ext ((field_is_discontinuous ? GeometryInfo<dim>::max_children_per_face
+                                   * GeometryInfo<dim>::faces_per_cell : 0),
+                                  FullMatrix<double>(finite_element.dofs_per_cell,
+                                                     finite_element.dofs_per_cell)),
+          local_matrices_ext_int ((field_is_discontinuous ? GeometryInfo<dim>::max_children_per_face
+                                   * GeometryInfo<dim>::faces_per_cell : 0),
+                                  FullMatrix<double>(finite_element.dofs_per_cell,
+                                                     finite_element.dofs_per_cell)),
+          local_matrices_ext_ext ((field_is_discontinuous ? GeometryInfo<dim>::max_children_per_face
+                                   * GeometryInfo<dim>::faces_per_cell : 0),
+                                  FullMatrix<double>(finite_element.dofs_per_cell,
+                                                     finite_element.dofs_per_cell)),
           local_rhs (finite_element.dofs_per_cell),
-          local_dof_indices (finite_element.dofs_per_cell)
+
+          assembled_matrices ((field_is_discontinuous ? GeometryInfo<dim>::max_children_per_face
+                               * GeometryInfo<dim>::faces_per_cell : 0), false),
+
+          local_dof_indices (finite_element.dofs_per_cell),
+          neighbor_dof_indices ((field_is_discontinuous ? GeometryInfo<dim>::max_children_per_face
+                                 * GeometryInfo<dim>::faces_per_cell : 0),
```

Comment:
Please break these lines (starting at line 368) differently so that an expression in either of the branches does not unnecessarily crosses lines. Maybe like
```
  field_is_discontinuous
  ?
  ...
  :
  0
```
?